src/compiler/scala/tools/util/PathResolver.scala

```diff
@@ -72,7 +72,8 @@ object PathResolver {
      */
     def sourcePathEnv       = envOrElse("SOURCEPATH", "")
 
-    def javaBootClassPath   = propOrElse("sun.boot.class.path", searchForBootClasspath)
+    //using propOrNone/getOrElse instead of propOrElse so that searchForBootClasspath is lazy evaluated
+    def javaBootClassPath   = propOrNone("sun.boot.class.path") getOrElse searchForBootClasspath
```

Comment:
It is already lazy.
```
def propOrElse(name: String, alt: => String)  = Option(System.getProperty(name)).getOrElse(alt)
```

Reply:
thanks, I didn't look that far. The scala 3 fork in 2013/14 means never knowing what is "ancient archaeology". I asked the same question to myself: modern implementation in scala 2 and 3 says "prefer by-name" (consider the string for an assert, the classic use case) but how often do you compute an alt string? I don't see other uses in dotty, so I wouldn't both porting it. It would not be a compat issue for dotty.

Reply:
> I don't see other uses in dotty, so I wouldn't bother porting it.

https://github.com/scala/scala3/pull/22255 Too late 😅